### ISSUE TITLE
Errors

### ISSUE THREAD

**Issue by JHarrison712** [Thetacz/nautobot-plugin-device-resources#2]
I'm not using docker, but when I install the pip install nautobot-device-resources and then echo nautobot-device-resources >> local_requirements.txt and then I modify the nautobot_config.py with PLUGINS = ["nautobot_device_resources"] then run nautobot-server post_upgrade I get some errors

Traceback (most recent call last):
  File "/opt/nautobot/bin/nautobot-server", line 8, in <module>
    sys.exit(main())
  File "/opt/nautobot/lib/python3.8/site-packages/nautobot/core/cli.py", line 54, in main
    run_app(
  File "/opt/nautobot/lib/python3.8/site-packages/nautobot/core/runner/runner.py", line 263, in run_app
    management.execute_from_command_line([runner_name, command] + command_args)
  File "/opt/nautobot/lib/python3.8/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/opt/nautobot/lib/python3.8/site-packages/django/core/management/__init__.py", line 395, in execute
    django.setup()
  File "/opt/nautobot/lib/python3.8/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/opt/nautobot/lib/python3.8/site-packages/django/apps/registry.py", line 114, in populate
    app_config.import_models()
  File "/opt/nautobot/lib/python3.8/site-packages/django/apps/config.py", line 301, in import_models
    self.models_module = import_module(models_module_name)
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 848, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/nautobot/lib/python3.8/site-packages/nautobot_device_resources/models.py", line 16, in <module>
    class DeviceResource(OrganizationalModel):
  File "/opt/nautobot/lib/python3.8/site-packages/nautobot_device_resources/models.py", line 58, in DeviceResource
    def cpu_threads(self) -> int | None:
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'

**Comment by jakubkrysl** [comment]
Hi, thanks for the report!
The reason is this: https://docs.python.org/3/library/stdtypes.html#types-union
TLDR: I use typing for Python 3.10 since its much easier and nicer to write `str | None` than `typing.Optional(str)` or using `typing.Union` in case of non-None value.
This should be the only issue with your Python3.8 and it might be worth fixing since Py3.9 has the same and it still will be supported for 2 years and 3 months from now.

I *might* have time to fix it in ~month, but if you want it faster, I'd be happy for any PRs :)
